### PR TITLE
Add extract-dialog-options.ui

### PR DESCRIPTION
**Pull request by rbuj**

### CODE REVIEW

File: src/dlg-extract.c [hunk 1]
```diff
@@ -276,131 +272,17 @@ static void
 files_entry_changed_cb (GtkWidget  *widget,
 			DialogData *data)
 {
-	if (! gtk_toggle_button_get_active (GTK_TOGGLE_BUTTON (data->e_files_radiobutton)))
-		gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (data->e_files_radiobutton), TRUE);
-}
+	if (! gtk_toggle_button_get_active (GTK_TOGGLE_BUTTON (GET_WIDGET ("file_pattern_radiobutton"))))
+		gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (GET_WIDGET ("file_pattern_radiobutton")), TRUE);}
```
Comment (sc0w) [inline src/dlg-extract.c]: ;**}**

**}** in new line is better

Comment (rbuj) [reply src/dlg-extract.c]: @sc0w Thanks

File: src/ui/extract-dialog-options.ui [hunk 2]
```diff
@@ -0,0 +1,229 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Generated with glade 3.22.1 -->
+<interface>
+  <requires lib="gtk+" version="3.0"/>
```
Comment (sc0w) [inline src/ui/extract-dialog-options.ui]: the minimum required for engrampa is gtk 3.22

**Comment by sc0w** [comment]
Glade says me:

```
[extra_widget:box5:box6:overwrite_checkbutton] Property 'Use Action Appearance' of object class 'Check Button' is deprecated
[extra_widget:box5:box6:overwrite_checkbutton] Property 'Horizontal alignment for child' of object class 'Button' is deprecated
[extra_widget:box5:box6:not_newer_checkbutton] Property 'Use Action Appearance' of object class 'Check Button' is deprecated
[extra_widget:box5:box6:not_newer_checkbutton] Property 'Horizontal alignment for child' of object class 'Button' is deprecated
[extra_widget:box5:box6:recreate_dir_checkbutton] Property 'Use Action Appearance' of object class 'Check Button' is deprecated
[extra_widget:box5:box6:recreate_dir_checkbutton] Property 'Horizontal alignment for child' of object class 'Button' is deprecated
[extra_widget:box2:box3:box4:file_pattern_radiobutton] Property 'Use Action Appearance' of object class 'Radio Button' is deprecated
[extra_widget:box2:box3:box4:file_pattern_radiobutton] Property 'Horizontal alignment for child' of object class 'Button' is deprecated
[extra_widget:box2:box3:selected_files_radiobutton] Property 'Use Action Appearance' of object class 'Radio Button' is deprecated
[extra_widget:box2:box3:selected_files_radiobutton] Property 'Horizontal alignment for child' of object class 'Button' is deprecated
[extra_widget:box2:box3:all_files_radiobutton] Property 'Use Action Appearance' of object class 'Radio Button' is deprecated
[extra_widget:box2:box3:all_files_radiobutton] Property 'Horizontal alignment for child' of object class 'Button' is deprecated
```

**Comment by sc0w** [comment]
```patch
diff --git a/src/ui/extract-dialog-options.ui b/src/ui/extract-dialog-options.ui
index 56a6672..7ab9fe1 100644
--- a/src/ui/extract-dialog-options.ui
+++ b/src/ui/extract-dialog-options.ui
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <!-- Generated with glade 3.22.1 -->
 <interface>
-  <requires lib="gtk+" version="3.0"/>
+  <requires lib="gtk+" version="3.22"/>
   <object class="GtkBox" id="extra_widget">
     <property name="visible">True</property>
     <property name="can_focus">False</property>
@@ -39,12 +39,11 @@
             <child>
               <object class="GtkRadioButton" id="all_files_radiobutton">
                 <property name="label" translatable="yes">_All files</property>
-                <property name="use_action_appearance">False</property>
                 <property name="visible">True</property>
                 <property name="can_focus">True</property>
                 <property name="receives_default">False</property>
+                <property name="halign">start</property>
                 <property name="use_underline">True</property>
-                <property name="xalign">0</property>
                 <property name="active">True</property>
                 <property name="draw_indicator">True</property>
               </object>
@@ -57,12 +56,11 @@
             <child>
               <object class="GtkRadioButton" id="selected_files_radiobutton">
                 <property name="label" translatable="yes">_Selected files</property>
-                <property name="use_action_appearance">False</property>
                 <property name="visible">True</property>
                 <property name="can_focus">True</property>
                 <property name="receives_default">False</property>
+                <property name="halign">start</property>
                 <property name="use_underline">True</property>
-                <property name="xalign">0</property>
                 <property name="active">True</property>
                 <property name="draw_indicator">True</property>
                 <property name="group">all_files_radiobutton</property>
@@ -81,12 +79,11 @@
                 <child>
                   <object class="GtkRadioButton" id="file_pattern_radiobutton">
                     <property name="label" translatable="yes">_Files:</property>
-                    <property name="use_action_appearance">False</property>
                     <property name="visible">True</property>
                     <property name="can_focus">True</property>
                     <property name="receives_default">False</property>
+                    <property name="halign">start</property>
                     <property name="use_underline">True</property>
-                    <property name="xalign">0</property>
                     <property name="active">True</property>
                     <property name="draw_indicator">True</property>
                     <property name="group">all_files_radiobutton</property>
@@ -163,12 +160,11 @@
             <child>
               <object class="GtkCheckButton" id="recreate_dir_checkbutton">
                 <property name="label" translatable="yes">Re-crea_te folders</property>
-                <property name="use_action_appearance">False</property>
                 <property name="visible">True</property>
                 <property name="can_focus">True</property>
                 <property name="receives_default">False</property>
+                <property name="halign">start</property>
                 <property name="use_underline">True</property>
-                <property name="xalign">0</property>
                 <property name="draw_indicator">True</property>
               </object>
               <packing>
@@ -178,14 +174,13 @@
               </packing>
             </child>
             <child>
-              <object class="GtkCheckButton" id="overwrite_checkbutton">
-                <property name="label" translatable="yes">Over_write existing files</property>
-                <property name="use_action_appearance">False</property>
+              <object class="GtkCheckButton" id="not_newer_checkbutton">
+                <property name="label" translatable="yes">Do not e_xtract older files</property>
                 <property name="visible">True</property>
                 <property name="can_focus">True</property>
                 <property name="receives_default">False</property>
+                <property name="halign">start</property>
                 <property name="use_underline">True</property>
-                <property name="xalign">0</property>
                 <property name="draw_indicator">True</property>
               </object>
               <packing>
@@ -194,15 +189,14 @@
                 <property name="position">1</property>
               </packing>
             </child>
-             <child>
-              <object class="GtkCheckButton" id="not_newer_checkbutton">
-                <property name="label" translatable="yes">Do not e_xtract older files</property>
-                <property name="use_action_appearance">False</property>
+            <child>
+              <object class="GtkCheckButton" id="overwrite_checkbutton">
+                <property name="label" translatable="yes">Over_write existing files</property>
                 <property name="visible">True</property>
                 <property name="can_focus">True</property>
                 <property name="receives_default">False</property>
+                <property name="halign">start</property>
                 <property name="use_underline">True</property>
-                <property name="xalign">0</property>
                 <property name="draw_indicator">True</property>
               </object>
               <packing>
@@ -211,7 +205,7 @@
                 <property name="position">1</property>
               </packing>
             </child>
-         </object>
+          </object>
           <packing>
             <property name="expand">False</property>
             <property name="fill">True</property>

**Comment by rbuj** [comment]
@sc0w done, thanks.

File: src/ui/extract-dialog-options.ui [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+                    <property name="expand">False</property>
+                    <property name="fill">True</property>
+                    <property name="position">0</property>
+                  </packing>
+                </child>
+                <child>
+                  <object class="GtkEntry" id="file_pattern_entry">
+                    <property name="visible">True</property>
+                    <property name="can_focus">True</property>
+                    <property name="tooltip_text" translatable="yes">example: *.o; *.bak</property>
```
Comment (sc0w) [inline src/ui/extract-dialog-options.ui]: before we have:

`example: *.txt; *.doc`

why this change?

Comment (rbuj) [reply src/ui/extract-dialog-options.ui]: @sc0w Thanks, I changed the filter pattern as you suggested.
